### PR TITLE
Fix warnings and add new flags

### LINKED ISSUE #1037: Forward port #1025 to the master branch
See: https://github.com/Exiv2/exiv2/pull/1025#issuecomment-540532771

### PR DESCRIPTION
**Pull request by piponazo**
Propagate changes from #1025 to master.

Fixes #1037

### CODE REVIEW

**Comment by piponazo** [comment]
Ey @D4N , this PR is waiting for your feedback. Any comment ? :)

File: src/actions.cpp [hunk 1]
```diff
@@ -1899,7 +1899,7 @@ namespace
         std::string tmp = "/tmp/";
 #endif
         char sCount[12];
-        sprintf(sCount, "_%d", ++count);
+        sprintf(sCount, "_%d", ++count); /// \todo replace by std::snprintf on master
```
Comment (D4N) [inline src/actions.cpp]: Well, this is for master, so we should replace it with `snprinft` ;-)

**Review by D4N — COMMENTED** [review]
Just one thing, why did you add commit f8158831250122c5471f25c66570b12632ffd645? If it is necessary, could you rebase it away with 10c33f7bf8ca53da3a52dd604235c6928d171660?

File: src/actions.cpp (hunk 1 above)
Comment (piponazo) [reply src/actions.cpp]: Done 😉

**Comment by piponazo** [comment]
> Just one thing, why did you add commit f815883? If it is necessary, could you rebase it away with 10c33f7?

@D4N I took car of squashing those 2 commits and fixing the TODO you pointed out.

**Comment by piponazo** [comment]
Note that this PR fixes partially #1039 by not modifying `CMAKE_CXX_FLAGS` anymore (although we keep modifying some `CMAKE_CXX_FLAGS_CONFIG` variables.

**Review by D4N — APPROVED** [review]
Thanks for taking care of this

**Pull request merged**